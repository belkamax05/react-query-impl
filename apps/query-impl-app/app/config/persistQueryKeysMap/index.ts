import persistQueryKeys, { PersistQueryKeysKey } from '../persistQueryKeys';

const persistQueryKeysMap = (<
  TKey extends PersistQueryKeysKey,
  TResult extends Record<TKey, true>
>(): TResult => {
  const result: Partial<TResult> = {};
  persistQueryKeys.forEach((queryKey) => {
    const jointKey = queryKey.join('.') as TKey;
    result[jointKey] = true as TResult[TKey];
  });
  return result as TResult;
})();

export default persistQueryKeysMap;
