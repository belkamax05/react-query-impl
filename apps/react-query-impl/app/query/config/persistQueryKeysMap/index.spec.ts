import persistQueryKeysMap from '.';

describe('persistQueryKeysMap', () => {
  it('should match snapshot', () => {
    expect(persistQueryKeysMap).toMatchSnapshot();
  });
});
