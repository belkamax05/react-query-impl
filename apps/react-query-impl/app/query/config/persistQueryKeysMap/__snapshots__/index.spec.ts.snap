// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`persistQueryKeysMap should match snapshot 1`] = `
{
  "todo": true,
}
`;
